capitulo2.md

```diff
@@ -55,55 +51,67 @@ Note que um literal de texto entre aspas duplas como `"my string"` não é do ti
+
+### As estruturas `while` e `for`
 
-As estruturas de controle `while` e `for` servem para repetir um bloco de código até que uma condição seja satisfeita. No exemplo abaixo, o programa pergunta se o usuário deseja aprender C++. Enquanto (`while`) a resposta não for `s` (sim) nem `n` (não), a mesma pergunta aparecerá para o usuário.
+As estruturas de controle `while` e `for` servem para repetir um bloco de código até que uma condição seja satisfeita. No exemplo abaixo, o programa pergunta se o usuário deseja aprender C++. Enquanto (`while`) a resposta não for `s` (sim) nem `n` (não), a mesma pergunta aparecerá para o usuário. Uma vez que o usuário der uma resposta válida, um `if` imprime a reação correspondente à resposta do usuário.
 
 ```
 #include <iostream>
 
 int main()
 {
-   char answer = 0;
+   char answer = '\0';
+
    while (answer != 's' && answer != 'n') {
        std::cout << "Você deseja aprender C++? [s/n]: ";
        std::cin >> answer;
    }
 
    if (answer == 's') {
-       std::cout << "Esse é o espírito!" << std::endl;
+       std::cout << "Esse é o espírito!\n";
    } else {
-       std::cout << "Que triste..." << std::endl;
+       std::cout << "Que triste...\n";
    }
-
-   return 0;
 }
 ```
 
-O laço “for” é apenas uma facilidade de execução, que acaba adicionando mais legibilidade ao código, colocando tanto a inicialização quanto o teste de continuidade e uma atualização de uma variável-alvo na mesma linha. Ou seja, tudo que é feito com um laço "for" pode também ser feito com um laço "while" e vice versa. A diferença é que, em alguns casos, a leitura do código fica mais natural com um ou com outro. No código abaixo, o laço for itera com valores de `i = a` até valor de `i = b`, somando 1 ao valor de `i` a cada iteração. Uma forma alternativa e mais compacta de escrever `i = i + 1` seria escrever `i++`.
+Em geral, usamos o laço `while` quando não existe uma definição clara de quantas vezes precisamos rodar o laço, apenas uma condição de continuidade que deve eventualmente tornar-se falsa.
+
+O laço `for` adiciona legibilidade ao código quando temos uma inicialização e um passo que deve ser executado em toda iteração do laço. Tudo que é feito com um laço `for` pode também ser feito com um laço `while`, e vice versa. A diferença entre eles é que, em alguns casos, a leitura do código fica mais natural com um ou com outro. Ao contrário do `while`, o `for` se presta mais quando sabemos exatamente quantas vezes precisamos repetir as operações. No código abaixo, o laço for itera com valores de `i = a` até valor de `i = b`, somando 1 ao valor de `i` a cada iteração. Uma forma alternativa e mais compacta de escrever `i = i + 1` seria escrever `++i`.
```

Comment:
Trocar (...) "código abaixo, o laço for itera com valores" (...)
Por (...) "código abaixo, o laço `for` itera com valores" (...)